NavMesh System

F3lifax opened the pull request:
Added basic navmesh system.
Used NavMeshPlus https://github.com/h8man/NavMeshPlus
Companion Follows Player

F3lifax committed:
commit

F3lifax committed:
Merge branch 'main' into TestFelix

F3lifax committed:
credits animation

Ian-kowalski committed:
Play credits animation and auto-return to main menu after completion

Enabled credits animation in the Credits scene and added logic to automatically return to the main menu once the animation finishes. Enhanced `CreditsManager` with animation playback and handling for a smoother transition experience.

F3lifax committed:
NAVAGENT STUFF

F3lifax committed:
Imported nav mesh package correctly (had to reimport)

F3lifax committed:
Add 2D companion follow script and update scene

Introduced CompanionFollow2D.cs for 2D NavMesh-based companion following, including jump and ground detection logic. Updated 'Test 1' scene to remove unused NavMeshModifier components, add new companion GameObjects, and integrate the new script. Added NavMeshPlus package to manifest for 2D navigation support.

F3lifax committed:
Reimported NavMeshPlus Package through Github URL

F3lifax committed:
Reimported the package AGAIN

navmeshplus name space is now accessible through main assembly references. Had to add name space in both main and the package's assembly references.

F3lifax committed:
Refactor scene and CompanionFollow2D for improved jumping

Updated the Test 1 scene to reorganize environment objects, adjust transforms, and remove unused GameObjects. Enhanced CompanionFollow2D with coyote time and jump buffer logic for more responsive jumping. Added serialized fields for jump settings, and improved ground check logic. Updated NavMesh assets and TimeManager settings for better physics consistency.

Ian-kowalski committed:
Merge branch 'main' into TestFelix

NotJezus reviewed (changes requested):
companionfollow2d line 10 to line 21, FollowState line 5 : do they need to be public?
movement is a empty script which should be deleted if not used.

Ian-kowalski commented on Assets/Scripts/movement.cs:
```diff
@@ -0,0 +1,16 @@
+    {
+        
+    }
+}
```
you can remove this script

Ian-kowalski commented on Assets/Scripts/State/CompanionFollow2D.cs:
```diff
@@ -0,0 +1,117 @@
+    [Header("Ground Settings")]
+    public float groundCheckRadius = 0.1f;
+    public Transform groundCheck;
+    public LayerMask groundLayer;
```
Does this need to be public?

Ian-kowalski commented on Assets/Scripts/State/FollowState.cs:
```diff
@@ -0,0 +1,21 @@
+using UnityEngine.AI;
+public class FollowState : MonoBehaviour
+{
+    public Transform target;
```
Make private

Ian-kowalski commented on Assets/Scripts/State/FollowState.cs:
```diff
@@ -0,0 +1,21 @@
+    public Transform target;
+    
+    private NavMeshAgent agent;
+    // Start is called once before the first execution of Update after the MonoBehaviour is created
```
pls remove comments

Ian-kowalski reviewed (commented):
Also, I merged it with main pls check if it works like it should

F3lifax committed:
Comments resolved

Made Publics to Privates (var) in CompanionFollow2D and Follow State.

Removed empty script.

Corrected pos for target point in player.

Made path calculation more efficient in CompanionFollow2D

F3lifax committed:
Companion no longer jitters when following player.

Changed when path to player updates.
Instead of updating path every 0.2 sec it updates when the player moves 1 unit.

Also added a fail safe to Recalculate path so it happens if the companion is too far from path.

Ian-kowalski committed:
Remove _Recovery folder and recovery scenes

Deleted the Assets/_Recovery folder along with its associated Unity scene files and metadata. This cleans up unused or temporary recovery assets from the project.

Ian-kowalski committed:
Merge branch 'main' into TestFelix

Ian-kowalski committed:
Add companion follow targeting and NavMesh setup

Introduces a TargetPoint child to the player prefab and updates CompanionFollow2D to dynamically assign and update its target. Adds NavMesh and NavMeshLink objects to the Test scene, instantiates the Miner prefab with companion follow logic, and updates PlayerManager to assign the TargetPoint to all companions. Also includes new NavMesh asset files for navigation.

Ian-kowalski committed:
Refactor grounded check in CompanionFollow2D

Removed the isGrounded field and CheckGrounded method, updating logic to use IsGrounded() directly for jump handling. This simplifies the code and avoids redundant state.

NotJezus reviewed (approved):
nice
